Move cpplint check to pre-commit

klssmith opened the pull request:
Currently, cpplint runs from https://github.com/pandas-dev/pandas/blob/master/ci/code_checks.sh
However, cpplint have a pre-commit hook which can be used.

- [x] closes https://github.com/MarcoGorelli/pyladies-sprint-March-2021/issues/2
- [ ] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

MarcoGorelli reviewed (changes requested):
Thanks @klssmith , looks good, there's just a conflict from #40277 - could you fetch upstream/master and merge please? If you could also make sure the hook is in alphabetical order (as the rest now are) that'd be great :+1:

klssmith committed:
Move cpplint check to pre-commit

Currently, cpplint runs from https://github.com/pandas-dev/pandas/blob/master/ci/code_checks.sh
However, cpplint have a pre-commit hook which can be used.

klssmith commented:
Thanks @MarcoGorelli I've changed the order and rebased now

MarcoGorelli reviewed (approved):
Awesome, thanks @klssmith, I tried running this locally and confirm it works as expected
